### PR TITLE
Switch to Falkordb python client

### PR DESCRIPTION
**Pull request by gkorland**
**Description:** replace the redis-py client with the native falkordb-py client
**Twitter handle:** g_korland

### CODE REVIEW

**Review by baskaryan — APPROVED** [review]

**Review by swilly22 — APPROVED** [review]

**Comment by baskaryan** [comment]
@gkorland needs to pass CI first, happy to help if you want to let me edit your fork. otherwise run
```python
cd libs/community
make format
make lint
```
to check lint issues

**Comment by gkorland** [comment]
@baskaryan thanks, I fixed the format error, fix the conflict and rebased the barnch

File: libs/community/langchain_community/graphs/falkordb_graph.py [hunk 1]
```diff
@@ -58,18 +58,21 @@ def __init__(
     ) -> None:
         """Create a new FalkorDB graph wrapper instance."""
         try:
-            import redis
-            from redis.commands.graph import Graph
+            from falkordb import FalkorDB
```
Comment (baskaryan) [inline libs/community/langchain_community/graphs/falkordb_graph.py:61]: this is a breaking change — thoughts on falling back to redis if falkordb isn't installed and raising a warning message that use of redis is deprecated?

Comment (gkorland) [reply libs/community/langchain_community/graphs/falkordb_graph.py]: done, pushed a fall back

**Comment by efriis** [comment]
Merging in #20229

For future PRs, you can let us edit your PR by raising from a personal branch (not org branch) to fix lint issues like this :) https://python.langchain.com/docs/contributing/faq/#how-do-i-allow-maintainers-to-edit-my-pr

**Pull request closed without merging**